crates/trigger/src/runtime_config/client_tls.rs

```diff
@@ -0,0 +1,39 @@
+use anyhow::Context;
+use rustls_pemfile::private_key;
+use std::io;
+use std::io::Cursor;
+use std::{fs, path::Path};
+
+#[derive(Debug, serde::Deserialize)]
+#[serde(rename_all = "snake_case", tag = "type")]
+pub struct ClientTlsOpts {
+    pub component_ids: Vec<String>,
+    pub hosts: Vec<String>,
+    pub custom_root_ca_file: Option<String>,
```

Comment:
Maybe `additional_root_ca_file` to clarify that it doesn't replace the default roots?

Reply:
updated the logic as per suggestion in a different comment (load default certs true by default unless custom ca provided, in which case it is disable by default. It can be forced to be enabled by explicit configuration).